File: 53-maximum-subarray/53-maximum-subarray.c
int maxSubArray(int* nums, int numsSize){
    int maxsum = nums[0];
    int cursum = 0;
    for(int i = 0; i < numsSize; ++i){
        cursum = cursum + nums[i];
        if(cursum > maxsum) maxsum = cursum;
        if(cursum < 0) cursum = 0;
    }  
    return maxsum;
}

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int curSum = 0;
        
        for(int i = 0; i < nums.length; i++){
            curSum += nums[i];
            if(curSum > maxSum) maxSum = curSum;
            if(curSum < 0) curSum = 0;
        }
        return maxSum;
    }
}